internal/util/jvm.go

```diff
@@ -47,5 +54,118 @@ func DetectJVMName(executor effect.Executor) (string, error) {
 			break
 		}
 	}
+
+	if err := scanner.Err(); err != nil {
+		return "", fmt.Errorf("unable to read cache ratio\n%w", err)
+	}
+
 	return "", nil
 }
+
+type SharedClassCache struct {
+	Name string
+	Path string
+
+	Executor effect.Executor
+	Logger   bard.Logger
+}
+
+type SharedClassCacheOptions struct {
+	Enabled       bool
+	SizeMB        int
+	NumIterations int
+	Trim          bool
+}
+
+func (scc SharedClassCache) GetFillRatio() (float64, error) {
+	sccOption := fmt.Sprintf("-Xshareclasses:name=%s,cacheDir=%s,printTopLayerStats", scc.Name, scc.Path)
+	args := []string{sccOption, "-version"}
+	buf := &bytes.Buffer{}
+	stderrBuf := &bytes.Buffer{}
+	err := scc.Executor.Execute(effect.Execution{
+		Command: "java",
+		Args:    args,
+		Stdout:  buf,
+		Stderr:  stderrBuf,
+	})
+	var exitError *exec.ExitError
+	if errors.As(err, &exitError) && exitError.ExitCode() != 1 {
+		return 0.0, fmt.Errorf("unable to get cache stats\n%w\n%s", err, stderrBuf.String())
+	}
+
+	r, err := regexp.Compile(`Cache is (\d+)% full`)
+	if err != nil {
+		return 0.0, err
+	}
+
+	scanner := bufio.NewScanner(buf)
+	for scanner.Scan() {
+		line := scanner.Text()
+		scc.Logger.Debugf("%s\n", line)
+		if matches := r.FindStringSubmatch(line); matches != nil {
+			v, err := strconv.ParseInt(matches[1], 10, 64)
+			if err != nil {
+				return 0.0, err
```

Comment:
```suggestion
				return 0.0, fmt.Errorf("unable to parse [%s] to int\n%w", matches[1], err)
```

I'm not sure everyone in Go-land would agree with me, but I will always wrap errors. I find it easier to debug things because it creates a basic stack trace, which I miss in Go.